### PR TITLE
Strip color codes when redirecting w/ ppaged()

### LINKED ISSUE #594: Strip escape codes when redirecting via ppaged()
Is there a reason ppaged doesn't strip escape codes when redirecting? Could look into creating a patch if it would be welcome.

### PR DESCRIPTION
**Pull request by ethanmcc**
Closes #594

### CODE REVIEW

File: cmd2/cmd2.py [hunk 1]
```diff
@@ -703,6 +703,9 @@ def ppaged(self, msg: str, end: str='\n', chop: bool=False) -> None:
                         else:
                             break
                     self.pipe_proc = None
+                elif self.redirecting and self.colors.lower() in (constants.COLORS_NEVER.lower(),
```
Comment (tleonhardt) [inline cmd2/cmd2.py]: I don’t think you need the elif.  I think you can just change the callnin the else below to one to self.decooorized_write.

Comment (ethanmcc) [reply cmd2/cmd2.py]: the elif should handle users with `colors = Always`

Comment (ethanmcc) [reply cmd2/cmd2.py]: oh nevermind, that code handles it. updating. also adjusted the docs.

File: cmd2/cmd2.py [hunk 2]
```diff
@@ -704,7 +704,7 @@ def ppaged(self, msg: str, end: str='\n', chop: bool=False) -> None:
                             break
                     self.pipe_proc = None
                 else:
-                    self.stdout.write(msg_str)
+                    self.decolorized_write(self.stdout, msg_str)
```
Comment (tleonhardt) [inline cmd2/cmd2.py]: This change looks good.

File: docs/unfreefeatures.rst [hunk 3]
```diff
@@ -146,19 +146,20 @@ The output methods in the previous section all honor the ``colors`` setting,
 which has three possible values:
 
 Never
-    poutput() and pfeedback() strip all ANSI escape sequences
+    poutput(), pfeedback(), and ppaged() strip all ANSI escape sequences
```
Comment (tleonhardt) [inline docs/unfreefeatures.rst]: Thanks for updating the docs!

File: tests/test_cmd2.py [hunk 4]
```diff
@@ -2109,6 +2109,24 @@ def test_ppaged(base_app):
     out = base_app.stdout.getvalue()
     assert out == msg + end
 
+def test_ppaged_strips_color_when_redirecting(base_app):
```
Comment (tleonhardt) [inline tests/test_cmd2.py]: And double thanks for adding unit tests 😀

**Comment by tleonhardt** [comment]
@kmvanbrunt @kotfu I’d appreciate a 2nd review if either of you have time.

**Comment by tleonhardt** [comment]
No need to recreate the PR.  I’ll merge it in late tonight.  Just waiting to see if other devs have feedback.

**Pull request merged**